Kernel/Process.cpp

```diff
@@ -4772,6 +4772,14 @@ int Process::sys$unveil(const Syscall::SC_unveil_params* user_params)
     if (path.value().is_empty() || path.value().characters()[0] != '/')
         return -EINVAL;
 
+    auto custody_or_error = VFS::the().resolve_path_without_veil(path.value(), root_directory());
```

Comment:
Hmm, I'm unsure what to think of this change.
With this change, we will now follow symbolic links when unveiling a path.
Previously, we would capture paths verbatim and stick with that.
The current implementation already has various other issues wrt symlinks, but what is the rationale for supporting symlinks here?

Reply:
@awesomekling That's what OpenBSD does AFAICT (by looking at the source code and using a quick test program).

If I unveil a file I would expect a symlink to it to be unveiled as well.

Here's the test program I used in OpenBSD: https://clbin.com/DAS0M.